test/e2e/s3/cross-account.sh

```diff
@@ -23,6 +29,32 @@ debug_msg "executing test: $service_name/$test_name"
 bucket_name="ack-test-smoke-$service_name-$AWS_ACCOUNT_ID_ALT-$RANDOM"
 resource_name="buckets/$bucket_name"
 
+# Create the testing namespace.
+# Note: this is step is not necessary.
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: $TESTING_NAMESPACE
+  annotations:
+    services.k8s.aws/default-region: "$AWS_REGION_ALT"
+    services.k8s.aws/owner-account-id: "$AWS_ACCOUNT_ID_ALT"
+EOF
+
+# Create the ack-role-account-map ConfigMap.
+# See CARM design: https://github.com/aws/aws-controllers-k8s/blob/main/docs/design/proposals/carm/cross-account-resource-management.md#storing-aws-role-arns
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ack-role-account-map
+  namespace: ack-system
+data:
+  "$AWS_ACCOUNT_ID_ALT": arn:aws:iam::$AWS_ACCOUNT_ID_ALT:role/$ASSUME_POLICY_ARN
+EOF
```

Comment:
I **think** that `kubectl apply` against a ConfigMap will no a strategic merge patch against the `data` field, right? In other words, by default it will not replace the `data` field with what you are passing here and instead **merge-update** the keys in the `data` map field.

Reply:
Thank you for pointing this, It’s a very interesting point! I think that It depends on how the previous configuration was created/updated. There are some cases where `kubectl apply` can completely replace a part of the `data` field (if not all of it).

Two rules i'm aware of:
- running `kubectl create` then `kubectl apply` = will do a **merge-update**
- running `kubectl apply` then another `kubectl apply` = will replace the previous `apply`-ed configuration

Reply:
I just tested this:

```bash
cat <<EOF | kubectl create -f -
apiVersion: v1
kind: ConfigMap
metadata:
  name: testmap
data:
  key1: value1
EOF

# This `kubectl apply` will perform a strategic merge
cat <<EOF | kubectl apply -f -
apiVersion: v1
kind: ConfigMap
metadata:
  name: testmap
data:
  key2: value2
EOF

# This `kubectl apply` will replace the previous `kubectl apply`-ed configuration
cat <<EOF | kubectl create -f -
apiVersion: v1
kind: ConfigMap
metadata:
  name: testmap
data:
  key3: value3
EOF
```

at the end the ConfigMap looks like:

```yaml
key1: value1
key3: value3
```

Reply:
@jaypipes I'm not sure if this is an issue, but I guess this can cause some unintended behavior if a user runs the smoke tests against a "real" cluster (since there are no rollout mechanisms for ConfigMap) Do you think that we should use `kubectl patch` instead ?

Reply:
I'm OK with this for now. We just need to be aware of this behaviour and perhaps do a safer technique in a followup patch. :)